mllib/src/main/scala/org/apache/spark/ml/classification/MultinomialLogisticRegression.scala

```diff
@@ -0,0 +1,611 @@
+           standardization. As a result, the objective history from optimizer is the same as the
+           one in the original space.
+         */
+        val arrayBuilder = mutable.ArrayBuilder.make[Double]
+        var state: optimizer.State = null
+        while (states.hasNext) {
+          state = states.next()
+          arrayBuilder += state.adjustedValue
+        }
+
+        if (state == null) {
+          val msg = s"${optimizer.getClass.getName} failed."
+          logError(msg)
+          throw new SparkException(msg)
+        }
+        bcFeaturesStd.destroy(blocking = false)
+
+        /*
+           The coefficients are trained in the scaled space; we're converting them back to
+           the original space.
+           Note that the intercept in scaled space and original space is the same;
+           as a result, no scaling is needed.
+         */
+        var interceptSum = 0.0
+        var coefSum = 0.0
+        val rawCoefficients = Vectors.fromBreeze(state.x)
+        val (coefMatrix, interceptVector) = rawCoefficients match {
+          case dv: DenseVector =>
+            val coefArray = Array.tabulate(numClasses * numFeatures) { i =>
+              val flatIndex = if ($(fitIntercept)) i + i / numFeatures else i
+              val featureIndex = i % numFeatures
+              val unscaledCoef = if (featuresStd(featureIndex) != 0.0) {
+                dv(flatIndex) / featuresStd(featureIndex)
+              } else {
+                0.0
+              }
+              coefSum += unscaledCoef
+              unscaledCoef
+            }
+            val interceptVector = if ($(fitIntercept)) {
+              Vectors.dense(Array.tabulate(numClasses) { i =>
+                val coefIndex = (i + 1) * numFeaturesPlusIntercept - 1
+                val intercept = dv(coefIndex)
+                interceptSum += intercept
+                intercept
+              })
+            } else {
+              Vectors.sparse(numClasses, Seq())
+            }
+            (new DenseMatrix(numClasses, numFeatures, coefArray, isTransposed = true),
+              interceptVector)
+          case sv: SparseVector =>
+            throw new IllegalArgumentException("SparseVector is not supported for coefficients")
+        }
+
+        /*
+          When no regularization is applied, the coefficients lack identifiability because
+          we do not use a pivot class. We can add any constant value to the coefficients and
+          get the same likelihood. So here, we choose the mean centered coefficients for
+          reproducibility. This method follows the approach in glmnet, described here:
+
+          Friedman, et al. "Regularization Paths for Generalized Linear Models via
+            Coordinate Descent," https://core.ac.uk/download/files/153/6287975.pdf
+         */
+        if ($(regParam) == 0.0) {
+          val coefficientMean = coefSum / (numClasses * numFeatures)
+          coefMatrix.update(_ - coefficientMean)
+        }
```

Comment:
wow, do they substrate the same value for all the different feature? my intuition will be substrating the mean of coef for each feature. Anyway, as long as we get the unique solution, and follow glmnet, it's great.